app/controllers/solutions_controller.rb

```diff
@@ -15,6 +15,11 @@ def index
     @solutions = @model.solutions
   end
 
+  def show
+    @solution = current_user.solutions.find_by!(challenge_id: allowed_params[:challenge_id])
```

Comment:
Could you check if the permitted params works here? I think you can use ti because you already permit challenge_id there

Reply:
permitted_params requires code and language, but this new request hasn't those.

Reply:
Ohhh you're right. Thinking on this, I prefer to use the challenge_id directly from the params objects. I believe we don't need to protect this value because everybody can see it in the url. Could yo drop the `allowed_params method` and just use `params[:challenge_id]` please